numpy/core/arrayprint.py

```diff
@@ -675,118 +675,121 @@ def _extendLine(s, line, word, line_width, next_line_prefix, legacy):
     line += word
     return s, line
 
+def _recursive_fmt(param, index, indent, curr_width):
+    """
+    Helper function for _formatArray, to recursively print array elements.
 
-def _formatArray(a, format_function, line_width, next_line_prefix,
-                 separator, edge_items, summary_insert, legacy):
-    """formatArray is designed for two modes of operation:
+    "param" are the values that are invariant under recursion, including
+    the array to be printed itself. index, indent and curr_width
+    are updated during recursion.
```

Comment:
A full up documentation string would be helpful here. I didn't worry about that for the backport, but it would be good to have it going forward.